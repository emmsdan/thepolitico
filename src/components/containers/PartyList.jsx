import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { getPartyRequestHandler } from '../../store/reducers/party';
import Loader from '../presentationals/Loader/Loader';
import PartyCard from '../presentationals/PartyCard/PartyCard';

class PartyList extends Component {
  componentDidMount() {
    this.props.getPartyRequestHandler('', 'parties');
  }

  render() {
    const parties = this.props.parties.parties.map((party, key) => {
      return <PartyCard {...party} key={key} />;
    });

    return (
      <>
        {this.props.parties.isLoading && (
          <Loader loader="show" text="Hi, am processing your input" />
        )}
        <div className="parties">
          <br /> <br />
          {parties}
        </div>
      </>
    );
  }
}

PartyList.propTypes = {
  getPartyRequestHandler: PropTypes.func.isRequired,
  parties: PropTypes.object,
  match: PropTypes.object,
  isLoading: PropTypes.bool,
};

const mapStateToProps = state => {
  return { parties: state.parties };
};

export default connect(
  mapStateToProps,
  { getPartyRequestHandler },
)(PartyList);
