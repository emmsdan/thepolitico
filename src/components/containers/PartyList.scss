.parties {
   display: flex;
   flex-wrap: wrap;
   width: 98%;
 }
