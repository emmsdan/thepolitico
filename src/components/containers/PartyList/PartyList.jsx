import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { getPartyRequestHandler } from '../../../store/reducers/party';
import Loader from '../../presentationals/Loader/Loader';
import PartyCard from '../../presentationals/PartyCard/PartyCard';

class PartyList extends Component {
  componentDidMount() {
    this.props.getPartyRequestHandler(this.props.partyId || '', 'parties');
  }

  render() {
    const parties = this.props.parties.parties.map((party, key) => {
      return (
        <PartyCard
          {...party}
          key={key}
          className={this.props.partyId ? 'party-card-2' : 'party-card-1'}
        />
      );
    });

    return (
      <>
        {this.props.parties.isLoading && <Loader loader="show" />}
        <div className="parties">
          <br /> <br />
          {parties}
        </div>
      </>
    );
  }
}

PartyList.propTypes = {
  getPartyRequestHandler: PropTypes.func.isRequired,
  parties: PropTypes.object,
  match: PropTypes.object,
  partyId: PropTypes.string,
  isLoading: PropTypes.bool,
};

const mapStateToProps = state => {
  return { parties: state.parties };
};

export default connect(
  mapStateToProps,
  { getPartyRequestHandler },
)(PartyList);
