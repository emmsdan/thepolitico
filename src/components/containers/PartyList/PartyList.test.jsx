import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import PartyList from './PartyList';

const props = {
  getPartyRequestHandler: jest.fn(),
  isLoading: false,
  parties: [{}, {}],
};

let store;
describe('PartyList', () => {
  beforeEach(() => {
    const mockStore = configureMockStore();
    store = mockStore({ ...props });
  });

  it('should render correctly', () => {
    const component = shallow(<PartyList store={store} />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
