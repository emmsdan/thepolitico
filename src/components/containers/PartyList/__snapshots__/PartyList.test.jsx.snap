// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartyList should render correctly 1`] = `
<ContextProvider
  value={null}
>
  <PartyList
    getPartyRequestHandler={[Function]}
    parties={
      Array [
        Object {},
        Object {},
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
